pkg/kubelet/kubelet.go

```diff
@@ -1715,6 +1727,42 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 	}
 }
 
+// syncStatus syncs pods statuses with the apiserver.
+func (kl *Kubelet) syncStatus() {
+	glog.V(3).Infof("Syncing pods status")
+
+	// Create the list of pods.
+	podFullNames := make(map[types.UID]string)
+	func() {
+		kl.podLock.Lock()
```

Comment:
You are not writing to kl.pods, so you can use GetPods() (which does RLock()) to get a copy of kl.pods.